File: c/day3/strongnumber.c
#include<stdio.h>
int factorial(int n){
    int fact=1;
    while(n>0){
        fact*=n;
        n--;
    }
    return fact;
}
int main(){
    int num,remainder,originalNum,sum=0;
    printf("enter a number:\n");
    scanf("%d",&num);;
    originalNum=num;
    while(originalNum!=0){
        remainder=originalNum%10;
        sum+=factorial(remainder);
        originalNum/=10;
    }
    if(sum==num){
        printf("%d is a strong number.\n",num);
    }
    else{
        printf("%d is not a strong number.\n",num);
    }
    return 0;
}

File: Java Interview/JavaInterviewtwoaugust/src/javainterview/StrongNumber.java
package javainterview;

import java.util.Scanner;

public class StrongNumber {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int num, originalNum, remainder, sum = 0;

		System.out.println("enter a number:");
		num = sc.nextInt();
		originalNum = num;
		while (num > 0) {
			remainder = num % 10;
			sum += fact(remainder);
			num = num / 10;

		}
		if (originalNum == sum) {
			System.out.println(originalNum + " is a strong number");
		} else {
			System.out.println(originalNum + " is not a strong number:");
		}

	}

	public static int fact(int num) {

		int fact = 1;
		for (int i = 1; i <= num; i++) {
			fact = fact * i;

		}
	
		return fact;
	}

}
